aspnetcore-webapi/GettingStartedASPNETMongoDB/Test/CourseControllerUnitTest.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Moq;
+using GettingStartedASPNETMongoDB.Controllers;
+using GettingStartedASPNETMongoDB.Interfaces;
+using GettingStartedASPNETMongoDB.Models;
+
+namespace Test
+{
+    public class CourseControllerUnitTest
+    {        
+        private Mock<ICourseService> _courseService;
+
+        private CourseController _courseController;
+
+        private Course _course;
+
+        public CourseControllerUnitTest()
+        {
+            Setup();
+        }
+
+        [Fact]
+        public async Task GivenCreate_WhenSentCorrectData_ThenReturnCreatedStatus()
+        {
+            //Act
+            var result = await _courseController.Create(_course);
+
+            //Assert
+            Assert.IsType<CreatedAtActionResult>(result);
+        }
+
+        [Fact]
+        public async Task GivenGetById_WhenSentCorrectId_ThenReturnStudentWithOkStatus()
+        {
+            //Arrange
+            var id = "1";
+
+            //Act
+            var result = await _courseController.GetById(id);
+
+            //Assert
+            var okResult = Assert.IsType<OkObjectResult>(result.Result);
+
+            Assert.IsType<Course>(okResult.Value);
+        }
```

Comment:
IN all the unit test method, we should have : 
(Given) some context
(When) some action is carried out
(Then) a particular set of observable consequences should obtain

Here, you are doing the action in the Given part and the context in the When part, it should be ther other way around, for example : 

GivenCreate_WhenSentCorrectData_ThenReturnCreatedStatus should be GivenCorrectData_WhenCreatingCourse_ThenReturnCreatedStatus.

This applies to the StudentControllerUnitTest as well.